exchanges/kucoin/kucoin_wrapper.go

```diff
@@ -0,0 +1,1349 @@
+		if err != nil {
+			return nil, err
+		}
+		return s.DeriveSubmitResponse(o)
+	case asset.Margin:
+		o, err := ku.PostMarginOrder(ctx, s.ClientOrderID, s.Side.Lower(), pairString, s.Type.Lower(), "", "", s.MarginMode, "", s.Price, s.Amount, s.TriggerPrice, s.Amount, 0, s.PostOnly, false, false, s.AutoBorrow)
+		if err != nil {
+			return nil, err
+		}
+		ret, err := s.DeriveSubmitResponse(o.OrderID)
+		if err != nil {
+			return nil, err
+		}
+		ret.BorrowSize = o.BorrowSize
+		ret.LoanApplyID = o.LoanApplyID
+		return ret, nil
+	default:
+		return nil, fmt.Errorf("%w %v", asset.ErrNotSupported, s.AssetType)
+	}
+}
+
+// ModifyOrder will allow of changing orderbook placement and limit to
+// market conversion
+func (ku *Kucoin) ModifyOrder(ctx context.Context, action *order.Modify) (*order.ModifyResponse, error) {
+	return nil, common.ErrFunctionNotSupported
+}
+
+// CancelOrder cancels an order by its corresponding ID number
+func (ku *Kucoin) CancelOrder(ctx context.Context, ord *order.Cancel) error {
+	err := ord.Validate(ord.StandardCancel())
+	if err != nil {
+		return err
+	}
+	switch ord.AssetType {
+	case asset.Spot, asset.Margin:
+		if ord.OrderID == "" && ord.ClientOrderID == "" {
+			return errors.New("either OrderID or ClientSuppliedOrderID must be specified")
+		}
+		if ord.OrderID != "" {
+			_, err = ku.CancelSingleOrder(ctx, ord.OrderID)
+		} else if ord.ClientOrderID != "" || ord.ClientID != "" {
+			if ord.ClientID != "" && ord.ClientOrderID == "" {
+				ord.ClientOrderID = ord.ClientID
+			}
+			_, err = ku.CancelOrderByClientOID(ctx, ord.ClientOrderID)
+		}
+		return err
+	case asset.Futures:
+		_, err := ku.CancelFuturesOrder(ctx, ord.OrderID)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// CancelBatchOrders cancels orders by their corresponding ID numbers
+func (ku *Kucoin) CancelBatchOrders(ctx context.Context, orders []order.Cancel) (order.CancelBatchResponse, error) {
+	return order.CancelBatchResponse{}, common.ErrFunctionNotSupported
+}
+
+// CancelAllOrders cancels all orders associated with a currency pair
+func (ku *Kucoin) CancelAllOrders(ctx context.Context, orderCancellation *order.Cancel) (order.CancelAllResponse, error) {
+	result := order.CancelAllResponse{}
+	err := orderCancellation.Validate()
+	if err != nil {
+		return result, err
+	}
+	var values []string
+	switch orderCancellation.AssetType {
+	case asset.Margin, asset.Spot:
+		var pairString string
+		if !orderCancellation.Pair.IsEmpty() {
+			pairString, err = ku.FormatSymbol(orderCancellation.Pair, orderCancellation.AssetType)
+			if err != nil {
+				return result, err
+			}
+		}
+		tradeType := ku.accountToTradeTypeString(orderCancellation.AssetType, orderCancellation.MarginMode)
+		values, err = ku.CancelAllOpenOrders(ctx, pairString, tradeType)
+		if err != nil {
+			return order.CancelAllResponse{}, err
+		}
+	case asset.Futures:
+		orderCancellation.Pair.Delimiter = ""
```

Comment:
This is because since the Futures instruments have no delimiters I can't set an empty string as Delimiter. Instead, I used the underscore delimiter and replaced the Underscore delimiter with an empty string whenever I wanted to use the currency Pair for outbound requests.

Reply:
That is not true. It will need to be formatted. I'll write some code for you to demonstrate this.

Reply:
Change your declaration of the following:
```
futures := currency.PairStore{
		RequestFormat: &currency.PairFormat{Uppercase: true, Delimiter: currency.UnderscoreDelimiter},
		ConfigFormat:  &currency.PairFormat{Uppercase: true, Delimiter: currency.UnderscoreDelimiter},
	}
```
to 
```
futures := currency.PairStore{
		RequestFormat: &currency.PairFormat{Uppercase: true},
		ConfigFormat:  &currency.PairFormat{Uppercase: true, Delimiter: currency.UnderscoreDelimiter},
	}
```
And then format pairs properly in wrapper functions